rust/main/chains/hyperlane-sealevel/src/tx_submitter.rs

```diff
@@ -54,20 +60,21 @@ impl TransactionSubmitter for RpcTransactionSubmitter {
         transaction: &Transaction,
         skip_preflight: bool,
     ) -> ChainResult<Signature> {
-        self.rpc_client
+        self.provider
+            .rpc_client()
             .send_transaction(transaction, skip_preflight)
             .await
     }
 
-    fn rpc_client(&self) -> Option<&SealevelRpcClient> {
-        Some(&self.rpc_client)
+    fn get_provider(&self) -> Option<&SealevelProvider> {
+        Some(&self.provider)
     }
 }
 
 /// A transaction submitter that uses the Jito API to submit transactions.
 #[derive(Debug, new)]
 pub struct JitoTransactionSubmitter {
-    rpc_client: SealevelRpcClient,
+    provider: Arc<SealevelProvider>,
```

Comment:
just assuming that you guys are starting to prefer Arc'd providers?

Reply:
Yeah, originally it was because we suspected providers were causing the high memory usage. But now with metrics out, it doesn't seem that was the case, so I'm cool with either (keeping it as Arc or removing it).